src/coreclr/vm/threads.h

```diff
@@ -5239,17 +5240,9 @@ class CoopTransitionHolder
     ~CoopTransitionHolder()
     {
         WRAPPER_NO_CONTRACT;
-        if (m_pFrame != NULL)
+        if (std::uncaught_exception())
```

Comment:
Is this change correct in all situations? Can this do a wrong thing with nested exception?

Also, what does this compile into? This executes on every JIT/EE interface call, so it is somewhat perf sensitive.

Reply:
Although, it should be ok for the current usage around JIT invocation, as we don't transitively call that from a destructor.

Reply:
Yeah I'd like to use `std::uncaught_exceptions()` once we do https://github.com/dotnet/runtime/issues/112419 (which it looks like we have buyoff on from our distro partners).

In the meantime I can add an assert in the constructor to ensure that this type isn't used in a case where a C++ exception is not already in flight (which it shouldn't be).

Reply:
>  I can add an assert in the constructor to ensure that this type isn't used in a case where a C++ exception is not already in flight (which it shouldn't be).

What guarantees that?

Reply:
> I am worried about JIT invocation to be called from under some other completely unrelated C++ exception handler. Is `std::uncaught_exceptions` going to be non-zero in that case?

If the code here is called from some other destructor triggered by an exception, the `std::uncaught_exceptions` will return non-zero even though the `~CoopTransitionHolder()` was called normally without exception triggering it. I've just verified that using a simple C++ app. So it would falsely indicate that the `~CoopTransitionHolder()` was called due to EH.